numpy/lib/function_base.py

```diff
@@ -2669,6 +2689,26 @@ class vectorize(object):
            <http://docs.scipy.org/doc/numpy/reference/c-api.generalized-ufuncs.html>`_.
     """
 
+    # __new__ is overriden as a clean way to provide decorator-with-keywords
+    # syntax while allowing 'vectorize' to remain as the type of vectorized
+    # functions.
+    def __new__(cls, pyfunc=None, *args, **kwargs):
+
+        def vectorize_decorator(pyfunc):
+            """ ``vectorize`` with presupplied keyword arguments. """
+            # prevent an edge case where None would produce another decorator
+            if not hasattr(pyfunc, '__call__'):
+                raise TypeError('the wrapped object must be callable')
+            return cls(pyfunc, *args, **kwargs)
+
+        # For backwards compatibility, subclasses must not use the overloaded
```

Comment:
to me this obfuscates what is otherwise an extemely simple control flow:

```python
if cls is vectorize and pyfunc is None:
    return vectorize_decorator
else:
    return object.__new__(cls)
```

which makes it easier to write the comment right above it that provides justification for the `type is` check.

I'll wait and see if the current style bothers anyone else.